### PR TITLE
`DataFrame.merge()` docstring improvement

### PR DESCRIPTION
**Pull request by thijs-obj**

### CODE REVIEW

File: bach/bach/dataframe.py [hunk 1]
```diff
@@ -2139,8 +2139,10 @@ def merge(
         :param right: DataFrame or Series to join on self
         :param how: supported values: {‘left’, ‘right’, ‘outer’, ‘inner’, ‘cross’}
         :param on: optional, column(s) or condition(s) to join left and right on.
-            If *on* is based on a condition (``SeriesBoolean``), the condition MUST make reference to both
-            nodes involved in the merge.
+            If *on* is based on a condition (:class:`SeriesBoolean`), then self and right must have different
```
Comment (KathiaBarahona) [inline bach/bach/dataframe.py:2142]: This statement : ``then self and right must have different
            base_nodes`` is not entirely true, both nodes must be referenced by the SeriesBoolean base node, but we don't care if they are the same. https://github.com/objectiv/objectiv-analytics/blob/2e6bd763deae4cadaa701b2054d9b138b26337f2/bach/bach/merge.py#L98-L102

Tho, a SeriesBoolean referencing twice to the same node is a very extreme edge case, can only imagine this happening if it was done manually

**Review by KathiaBarahona — APPROVED** [review]
lgtm, just a small comment

File: bach/bach/dataframe.py (hunk 1 above)
Comment (thijs-obj) [reply bach/bach/dataframe.py:2142]: A few lines above we raise an exception if `left.base_node == right.base_node` https://github.com/objectiv/objectiv-analytics/blob/2e6bd763deae4cadaa701b2054d9b138b26337f2/bach/bach/merge.py#L95-L96

Comment (thijs-obj) [reply bach/bach/dataframe.py:2142]: I understood that correct, right? @KathiaBarahona

Comment (KathiaBarahona) [reply bach/bach/dataframe.py:2142]: Oh, then entirely ignore me haha

**Pull request merged**